app/src/main/java/protect/card_locker/LoyaltyCardEditActivity.java

```diff
@@ -182,12 +206,29 @@ private void extractIntentFields(Intent intent) {
                 + ", updateLoyaltyCard=" + updateLoyaltyCard);
     }
 
+
+
     @Override
     public void onSaveInstanceState(@NonNull Bundle savedInstanceState) {
         super.onSaveInstanceState(savedInstanceState);
         tabs = findViewById(R.id.tabs);
         savedInstanceState.putInt(STATE_TAB_INDEX, tabs.getSelectedTabPosition());
         savedInstanceState.putParcelable(STATE_TEMP_CARD, tempLoyaltyCard);
+        savedInstanceState.putInt(STATE_REQUESTED_IMAGE, mRequestedImage);
+        Object cardImageFrontObj = cardImageFront.getTag();
+        if (mFrontImageUnsaved && (cardImageFrontObj instanceof Bitmap) && Utils.saveTempImage(this, (Bitmap)cardImageFrontObj, TEMP_UNSAVED_FRONT_IMAGE_NAME, TEMP_UNSAVED_IMAGE_FORMAT) != null){
+            savedInstanceState.putInt(STATE_FRONT_IMAGE_UNSAVED, 1);
+        }else{
+            savedInstanceState.putInt(STATE_FRONT_IMAGE_UNSAVED, 0);
+        }
+        Object cardImageBackObj = cardImageBack.getTag();
+        if (mBackImageUnsaved && (cardImageBackObj instanceof Bitmap) && Utils.saveTempImage(this, (Bitmap)cardImageBackObj, TEMP_UNSAVED_BACK_IMAGE_NAME, TEMP_UNSAVED_IMAGE_FORMAT) != null){
+            savedInstanceState.putInt(STATE_BACK_IMAGE_UNSAVED, 1);
+        }else{
+            savedInstanceState.putInt(STATE_BACK_IMAGE_UNSAVED, 0);
+        }
```

Comment:
Oh nice this fixes a bug where rotating the screen actually makes a set picture disappear that I wasn't aware of (just created #568 to keep track of that bug).

I did notice though that with your new code if I have a saved card that already had an image, edit the card, choose "Remove image" and then rotate the screen the removed image returns. So it saves state slightly too long. But otherwise thank you so much for improving an issue I didn't even know existed!